vendor/go.mozilla.org/cose/sign_verify_test.go

```diff
@@ -0,0 +1,212 @@
+package cose
+
+import (
+	"crypto"
+	"crypto/dsa"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"errors"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"math/rand"
+	"testing"
+)
+
+func TestSignErrors(t *testing.T) {
+	assert := assert.New(t)
+
+	randReader := rand.New(rand.NewSource(int64(0)))
+
+	msg := NewSignMessage([]byte("payload to sign"))
+
+	ecdsaPrivateKey := ecdsa.PrivateKey{
+		PublicKey: ecdsa.PublicKey{
+			Curve: elliptic.P256(),
+			X:     FromBase64Int("usWxHK2PmfnHKwXPS54m0kTcGJ90UiglWiGahtagnv8"),
+			Y:     FromBase64Int("IBOL-C3BttVivg-lSreASjpkttcsz-1rb7btKLv8EX4"),
+		},
+		D: FromBase64Int("V8kgd2ZBRuh2dgyVINBUqpPDr7BOMGcF22CQMIUHtNM"),
+	}
+
+	dsaPrivateKey := dsa.PrivateKey{
+		PublicKey: dsa.PublicKey{
+			Parameters: dsa.Parameters{
+				P: FromBase64Int("A9B5B793FB4785793D246BAE77E8FF63CA52F442DA763C440259919FE1BC1D6065A9350637A04F75A2F039401D49F08E066C4D275A5A65DA5684BC563C14289D7AB8A67163BFBF79D85972619AD2CFF55AB0EE77A9002B0EF96293BDD0F42685EBB2C66C327079F6C98000FBCB79AACDE1BC6F9D5C7B1A97E3D9D54ED7951FEF"),
+				Q: FromBase64Int("E1D3391245933D68A0714ED34BBCB7A1F422B9C1"),
+				G: FromBase64Int("634364FC25248933D01D1993ECABD0657CC0CB2CEED7ED2E3E8AECDFCDC4A25C3B15E9E3B163ACA2984B5539181F3EFF1A5E8903D71D5B95DA4F27202B77D2C44B430BB53741A8D59A8F86887525C9F2A6A5980A195EAA7F2FF910064301DEF89D3AA213E1FAC7768D89365318E370AF54A112EFBA9246D9158386BA1B4EEFDA"),
+			},
+			Y: FromBase64Int("32969E5780CFE1C849A1C276D7AEB4F38A23B591739AA2FE197349AEEBD31366AEE5EB7E6C6DDB7C57D02432B30DB5AA66D9884299FAA72568944E4EEDC92EA3FBC6F39F53412FBCC563208F7C15B737AC8910DBC2D9C9B8C001E72FDC40EB694AB1F06A5A2DBD18D9E36C66F31F566742F11EC0A52E9F7B89355C02FB5D32D2"),
+		},
+		X: FromBase64Int("5078D4D29795CBE76D3AACFE48C9AF0BCDBEE91A"),
+	}
+
+	signer, err := NewSigner(&ecdsaPrivateKey)
+	assert.Nil(err, fmt.Sprintf("Error creating signer %s", err))
+
+	opts := SignOpts{
+		HashFunc: crypto.SHA256,
+		GetSigner: func(index int, signature Signature) (Signer, error) {
+			return *signer, ErrNoSignerFound
+		},
+	}
+
+	sig := NewSignature()
+	sig.Headers.Protected[algTag] = -41 // RSAES-OAEP w/ SHA-256 from [RFC8230]
```

Comment:
Maybe I'm missing something but I don't see tests with intermediate certs in the protected body header. Note that the Firefox code expects there always to be a (potentially empty) list of intermediate certs.

Reply:
Yeah I still need to write Fx/XPI specific tests and checks.